Uh, I don't think it's related to my PR:

```
protocols/http.c:68:80: error: unused parameter ‘ndpi_struct’ [-Werror=unused-parameter]
   68 | static void ndpi_set_binary_data_transfer(struct ndpi_detection_module_struct *ndpi_struct,
      |      
```

--- a/src/lib/ndpi_main.c
+++ b/src/lib/ndpi_main.c
@@ -11442,9 +11442,8 @@ void* ndpi_memmem(const void* haystack, size_t haystack_len, const void* needle,
     return memchr(haystack, *(int*)needle, haystack_len);
   }
 
-  for (const u_int8_t* h = haystack; h <= (const u_int8_t*)haystack + 
-       haystack_len - needle_len; ++h)
-  {
+  const u_int8_t* h = NULL;
+  for (h = haystack; h <= (const u_int8_t*)haystack+haystack_len-needle_len; ++h) {
     if (!memcmp(h, needle, needle_len))
       return (void*)h;
   }

--- a/src/lib/ndpi_main.c
+++ b/src/lib/ndpi_main.c
@@ -11443,8 +11443,9 @@ void* ndpi_memmem(const void* haystack, size_t haystack_len, const void* needle,
   }
 
   const u_int8_t* h = NULL;
+  const u_int8_t* n = (const u_int8_t*)needle;
   for (h = haystack; h <= (const u_int8_t*)haystack+haystack_len-needle_len; ++h) {
-    if (!memcmp(h, needle, needle_len))
+    if (*h == n[0] && !memcmp(h, needle, needle_len))
       return (void*)h;
   }
 

